### PR TITLE
feat(amazon-bedrock,google-vertex-anthropic): Anthropic Structured Tool Call Ports

### PR DESCRIPTION
**Pull request by Und3rf10w**
## Background

The Anthropic provider recently added support for combining tool calling with structured outputs in commit cf4e2a9d0 (#9793). This feature allows developers to use tools (like web search, weather lookup, etc.) alongside structured JSON output schemas, enabling multi-step workflows with structured final outputs.

However, this capability was not ported to the Amazon Bedrock and Google Vertex Anthropic providers

## Summary

This PR ports the structured tool call support from the Anthropic provider to both Amazon Bedrock and Google Vertex Anthropic providers.

### Amazon Bedrock Provider Changes

**Core Implementation:**
- Removed the incorrect warning that prevented using tools with JSON response format (bedrock-chat-language-model.ts:119-128)
- Fixed tool choice to use `{ type: 'required' }` instead of `{ type: 'tool', toolName: 'json' }` when combining tools with structured outputs (bedrock-chat-language-model.ts:133-134)
- Added `isJsonResponseFromTool` parameter to `mapBedrockFinishReason` function to correctly map finish reason from `tool_use` to `stop` when the JSON tool is used (map-bedrock-finish-reason.ts)
- Updated both `doGenerate` and `doStream` methods to pass `usesJsonResponseFromTool` flag to the finish reason mapper

**Testing:**
- Replaced the old warning test with two comprehensive test suites:
  - `json schema response format with json tool response` - Tests when only the JSON tool is called
  - `json schema response format with other tool response` - Tests when regular tools are called alongside JSON schema
- Tests verify tool configuration, response format, finish reasons, and provider metadata

**Examples:**
- Added [Generate Text Example](examples/ai-core/src/generate-text/amazon-bedrock-anthropic-output-array-tools.ts)
- Added [Stream Text Example](examples/ai-core/src/stream-text/amazon-bedrock-anthropic-output-array-tools.ts)

### Google Vertex Anthropic Provider Changes

Since this provider inherits from the Anthropic provider through `AnthropicMessagesLanguageModel`, it automatically gains the feature. Added:

**Testing:**
- Added test verifying the feature works correctly through inheritance (google-vertex-anthropic-provider.test.ts:181-198)

**Examples:**
- Added [Generate Text Example](examples/ai-core/src/generate-text/google-vertex-anthropic-output-array-tools.ts)
- Added [Stream Text Example](examples/ai-core/src/stream-text/google-vertex-anthropic-output-array-tools.ts)

## Manual Verification

Run the following examples to verify the feature works end-to-end:

**Amazon Bedrock:**
```bash
# Generate text with tools and structured output
pnpm tsx examples/ai-core/src/generate-text/amazon-bedrock-anthropic-output-array-tools.ts

# Stream text with tools and structured output
pnpm tsx examples/ai-core/src/stream-text/amazon-bedrock-anthropic-output-array-tools.ts
```

**Google Vertex Anthropic:**
```bash
# Generate text with tools and structured output
pnpm tsx examples/ai-core/src/generate-text/google-vertex-anthropic-output-array-tools.ts

# Stream text with tools and structured output
pnpm tsx examples/ai-core/src/stream-text/google-vertex-anthropic-output-array-tools.ts
```

Expected behavior:
- The model can use the weather tool to fetch information
- The final output is a structured JSON array matching the schema
- No warnings about tools being ignored with JSON response format
- Finish reason is correctly set to `stop` when JSON tool is used, and `tool-calls` when regular tools are used

## Checklist

- [x] Tests have been added / updated (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] A _patch_ changeset for relevant packages has been added (for bug fixes / features - run `pnpm changeset` in the project root)
- [x] I have reviewed this pull request (self-review)

## Related Issues

- Related to #9789 (original issue for Anthropic provider)
  - Implements the same feature for Amazon Bedrock and Google Vertex Anthropic providers

### CODE REVIEW

File: packages/amazon-bedrock/src/bedrock-chat-language-model.ts [hunk 1]
```diff
@@ -588,24 +578,17 @@ export class BedrockChatLanguageModel implements LanguageModelV3 {
...
-                if (!usesJsonResponseTool) {
-                  controller.enqueue({
-                    type: 'text-start',
-                    id: String(blockIndex),
-                  });
-                }
-              }
-
-              // when a json response tool is used, we don't emit text events
-              if (!usesJsonResponseTool) {
```
Comment (lgrammel) [inline packages/amazon-bedrock/src/bedrock-chat-language-model.ts:623]: wont this removal introduce bugs when text messages are generated before the tool call?

File: packages/amazon-bedrock/src/bedrock-chat-language-model.ts [hunk 2]
```diff
@@ -329,16 +317,13 @@ export class BedrockChatLanguageModel implements LanguageModelV3 {
...
 
     // map response content to content array
     for (const part of response.output.message.content) {
       // text
       if (part.text) {
-        // when a json response tool is used, the tool call is returned as text,
-        // so we ignore the text content:
-        if (!usesJsonResponseTool) {
-          content.push({ type: 'text', text: part.text });
-        }
```
Comment (lgrammel) [inline packages/amazon-bedrock/src/bedrock-chat-language-model.ts:363]: potential bug source, see below

File: packages/amazon-bedrock/src/bedrock-chat-language-model.ts [hunk 3]
```diff
@@ -619,15 +602,13 @@ export class BedrockChatLanguageModel implements LanguageModelV3 {
                     id: String(blockIndex),
                   });
                 } else if (contentBlock.type === 'text') {
-                  // when a json response tool is used, we don't emit text events
```
Comment (lgrammel) [inline packages/amazon-bedrock/src/bedrock-chat-language-model.ts:643]: same

**Comment by lgrammel** [comment]
ideally introduce tests with fixtures and snapshots like we have in the anthropic provider

File: packages/amazon-bedrock/src/bedrock-chat-language-model.ts (hunk 1 above)
Comment (Und3rf10w) [reply packages/amazon-bedrock/src/bedrock-chat-language-model.ts:623]: I added some tests to cover potential edge cases like this in 0f288748aa47c8f24e890a5ee22dfdd52934916d, but either I'm doing this wrong and it's not catching them, or it's being accounted for by below:

Here we're adding all `text` parts to the `content,` regardless of `usesJsonResponseTool`:

https://github.com/vercel/ai/blob/0f288748aa47c8f24e890a5ee22dfdd52934916d/packages/amazon-bedrock/src/bedrock-chat-language-model.ts#L323-L326

Then, while processing tool calls:
- If it's the JSON response tool, it sets `isJsonResponseFromTool = true` and adds the tool input as text content
- Otherwise, it adds a regular `tool-call`

https://github.com/vercel/ai/blob/0f288748aa47c8f24e890a5ee22dfdd52934916d/packages/amazon-bedrock/src/bedrock-chat-language-model.ts#L362-L369

This block only skips tool-specific events (not text events) when it's a JSON response tool:

https://github.com/vercel/ai/blob/0f288748aa47c8f24e890a5ee22dfdd52934916d/packages/amazon-bedrock/src/bedrock-chat-language-model.ts#L712-L719

- The `usesJsonResponseTool` flag only affects:
  - Converting the JSON tool's output to text format
  - Changing the finish reason from `tool-calls` to `stop`
  - Adding metadata flags

Are there other potential edge cases I might be missing to cover, or would you prefer I rewrote this in a more readable fashion?

**Comment by Und3rf10w** [comment]
> ideally introduce tests with fixtures and snapshots like we have in the anthropic provider

1dde227e497e9a019bbd09992d9c20696e9656a2 moved existing tests to fixtures

File: packages/amazon-bedrock/src/bedrock-chat-language-model.ts (hunk 1 above)
Comment (Und3rf10w) [reply packages/amazon-bedrock/src/bedrock-chat-language-model.ts:623]: Although I've tested this live now finally and I'm having trouble getting a text part. I'll come back to this in a few

Comment (Und3rf10w) [reply packages/amazon-bedrock/src/bedrock-chat-language-model.ts:623]: @lgrammel 

I have added more test coverage just in case because I was a bit worried about this. I cannot for the life of me find a case where that happens with the code currently.

This test, in theory, should cover that case:

https://github.com/vercel/ai/blob/c439b7664f6ebc139bf672daa6f8718233b377fa/packages/amazon-bedrock/src/bedrock-chat-language-model.test.ts#L1630-L1703

If you have any suggestions on how else to account for that, happy to make adjustments.

Also if you have any guidance on how to back port this to v5, I would love to do that as well.

**Review by gr2m — APPROVED** [review]
Thank you @Und3rf10w for another great pull request. Apologies it took us a while to get it merged
